dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java

```diff
@@ -94,18 +98,32 @@ public Relationship create(Context context, Relationship relationship) throws SQ
     }
 
     @Override
-    public void updatePlaceInRelationship(Context context, Relationship relationship, boolean isCreation)
+    public void updatePlaceInRelationship(Context context, Relationship relationship)
         throws SQLException, AuthorizeException {
         Item leftItem = relationship.getLeftItem();
+        // Max value is used to ensure that these will get added to the back of the list and thus receive the highest
+        // (last) place as it's set to a -1 for creation
+        if (relationship.getLeftPlace() == -1) {
+            relationship.setLeftPlace(Integer.MAX_VALUE);
+        }
+        Item rightItem = relationship.getRightItem();
+        if (relationship.getRightPlace() == -1) {
+            relationship.setRightPlace(Integer.MAX_VALUE);
+        }
         List<Relationship> leftRelationships = findByItemAndRelationshipType(context,
                                                                              leftItem,
                                                                              relationship.getRelationshipType(), true);
-        Item rightItem = relationship.getRightItem();
         List<Relationship> rightRelationships = findByItemAndRelationshipType(context,
                                                                               rightItem,
                                                                               relationship.getRelationshipType(),
                                                                               false);
 
+        if (leftRelationships.contains(relationship)) {
+            leftRelationships.remove(relationship);
+        }
+        if (rightRelationships.contains(relationship)) {
+            rightRelationships.remove(relationship);
```

Comment:
These if statements should be preceded by a comment to describe why we are removing the relationship from the list.  I don't understand the logic here.  Could we add some Unit Tests to prove out this is working as expected?

Reply:
These relationships here aren't removed from the actual list, they're only removed in this temporary list so that we can perform a proper place calculation further on in this method. I do agree that it was a bit unclear, so I'll add a comment stating this. This is particularly useful incase this method is called for a delete use case, so that we can update the relationships' places properly without the relationship given interfering with the place calculation

Reply:
This has been clarified using inline comments